pkg/reconciler/emit_metrics.go

```diff
@@ -11,12 +11,17 @@ func (r *Reconciler) emitMetrics(pr *tektonv1.PipelineRun) error {
 	gitProvider := pr.GetAnnotations()[keys.GitProvider]
```

Comment:
some sanity unit test, maybe ?

copilot gave me that to start with

```go
import (
    "testing"
    tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func TestEmitMetrics(t *testing.T) {
    tests := []struct {
        name        string
        annotations map[string]string
        wantErr     bool
    }{
        {
            name: "github app",
            annotations: map[string]string{
                keys.GitProvider:     "github",
                keys.EventType:       "push",
                keys.InstallationID:  "123",
            },
            wantErr: false,
        },
        {
            name: "unsupported provider",
            annotations: map[string]string{
                keys.GitProvider: "unsupported",
                keys.EventType:   "push",
            },
            wantErr: true,
        },
        // Add more test cases as needed
    }

    for _, tt := range tests {
        t.Run(tt.name, func(t *testing.T) {
            r := &Reconciler{
                metrics: &MockMetrics{},
            }
            pr := &tektonv1.PipelineRun{
                ObjectMeta: metav1.ObjectMeta{
                    Annotations: tt.annotations,
                },
            }

            err := r.emitMetrics(pr)
            if (err != nil) != tt.wantErr {
                t.Errorf("emitMetrics() error = %v, wantErr %v", err, tt.wantErr)
            }
        })
    }
}
```

Reply:
added tests